packages/lodestar/src/network/reqresp/request/index.ts

```diff
@@ -87,20 +86,17 @@ export async function sendRequest<T extends phase0.ResponseBody | phase0.Respons
     logger.debug("Req  sending request", {...logCtx, requestBody} as Context);
 
     // Spec: The requester MUST close the write side of the stream once it finishes writing the request message
-    // Impl: stream.sink should be closed automatically by js-libp2p-mplex when piped source returns
+    // Impl: stream.sink is closed automatically by js-libp2p-mplex when piped source is exhausted
 
     // REQUEST_TIMEOUT: Non-spec timeout from sending request until write stream closed by responder
+    // Note: libp2p.stop() will close all connections, so not necessary to abort this pipe on parent stop
     await withTimeout(
-      async (timeoutAndParentSignal) =>
-        await pipe(
-          abortSource(requestEncode(config, method, encoding, requestBody), timeoutAndParentSignal),
-          stream.sink
-        ),
+      async () => await pipe(requestEncode(config, method, encoding, requestBody), stream.sink),
       REQUEST_TIMEOUT,
       signal
     ).catch((e) => {
-      // Must close the stream read side (stream.source) manually
-      stream.close();
+      // Must close the stream read side (stream.source) manually AND the write side
+      stream.reset();
```

Comment:
I think we want to `abort` here?

Reply:
I end up using reset because abort is not typed in your libp2p types and has the same exact effect with the difference that abort logs the error to the libp2p internal debugger.

Reply:
I think they're not exactly the same, see https://github.com/libp2p/js-libp2p-mplex/blob/master/src/stream.js#L101-L106
I think we should still use `abort` here.
For now, we can update our types to match (or ts-ignore :shrug: ). Slightly longer term, do this: #2167